r-package/grf/R/local_linear_forest.R

```diff
@@ -68,48 +79,124 @@
+                                honesty.prune.leaves = TRUE,
+                                alpha = 0.05,
+                                imbalance.penalty = 0,
+                                ci.group.size = 2,
+                                tune.parameters = "none",
+                                tune.num.trees = 50,
+                                tune.num.reps = 100,
+                                tune.num.draws = 1000,
+                                num.threads = NULL,
+                                seed = runif(1, 0, .Machine$integer.max)) {
+
+  validate_X(X)
+  validate_sample_weights(sample.weights, X)
+  Y <- validate_observations(Y, X)
+  clusters <- validate_clusters(clusters, X)
+  samples.per.cluster <- validate_equalize_cluster_weights(equalize.cluster.weights, clusters, sample.weights)
+  num.threads <- validate_num_threads(num.threads)
+
+  ll.split.variables <- validate_ll_vars(ll.split.variables, ncol(X))
+  ll.split.lambda <- validate_ll_lambda(ll.split.lambda)
+  ll.split.cutoff <- validate_ll_cutoff(ll.split.cutoff, nrow(X))
+
+  all.tunable.params <- c("sample.fraction", "mtry", "min.node.size", "honesty.fraction",
+                          "honesty.prune.leaves", "alpha", "imbalance.penalty")
+
+  data <- create_data_matrices(X, outcome = Y, sample.weights = sample.weights)
+
+  args <- list(num.trees = num.trees,
+               clusters = clusters,
+               samples.per.cluster = samples.per.cluster,
+               sample.fraction = sample.fraction,
+               mtry = mtry,
+               min.node.size = min.node.size,
+               honesty = honesty,
+               honesty.fraction = honesty.fraction,
+               honesty.prune.leaves = honesty.prune.leaves,
+               alpha = alpha,
+               imbalance.penalty = imbalance.penalty,
+               ci.group.size = ci.group.size,
+               num.threads = num.threads,
+               seed = seed)
+  if (ll.splits & ll.split.cutoff > 0) {
+    # find overall beta
+    J = diag(ncol(X) + 1)
+    J[1,1] = 0
+    D = cbind(1, X)
+    overall.beta = solve(t(D) %*% D + ll.split.lambda*J) %*% t(D) %*% Y
+
+    # update arguments with LLF parameters
+    args <- c(args, list(weight.penalty = ll.split.weight.penalty,
+                         split.lambda = ll.split.lambda,
+                         ll.split.variables = ll.split.variables,
+                         ll.split.cutoff = ll.split.cutoff,
+                         overall.beta = overall.beta))
+  } else if (ll.splits) {
+    # update arguments with LLF parameters
+    args <- c(args, list(weight.penalty = ll.split.weight.penalty,
+                         split.lambda = ll.split.lambda,
+                         ll.split.variables = ll.split.variables,
+                         ll.split.cutoff = ll.split.cutoff,
+                         overall.beta = vector(mode = "numeric", length = 0)))
+  } else {
+    args <- c(args, compute.oob.predictions = FALSE)
```

Comment:
shouldn't this be removed?

Reply:
In this case, we call the standard regression forest training because users haven't requested local linear splits - so we have to define compute.oob.predictions = FALSE for that call.